docs/node/migrations/overview.mdx

```diff
@@ -0,0 +1,26 @@
+---
+sidebar_position: 1
+sidebar_label: "Overview"
+id: migrations-overview
+title: Migrations
+description: Overview of Kwil's migration system
+slug: /node/migrations/overview
+---
+
+This section provides an overview of the migration system in Kwil. Migrations are used to upgrade the network to a new version or to a new network with minimal downtime. Depending on the nature of the upgrade, migrations can be coordinated or uncoordinated.
+
+## Rolling upgrades
+
+Whenever network has to be upgraded to a newer versions of `kwild` binaries which doesn't have any breaking changes to the consensus rules, the upgrade can be done in a rolling fashion. In a rolling upgrade, the validators can upgrade their nodes one by one without stopping the network and still ensure that the nodes agree on the app hash for the block.
+
+## Coordinated Upgrades
+
+Whenever there are breaking changes to the consensus rules the network has to be upgraded in a coordinated fashion. Kwil currently supports below two types of coordinated upgrades differing in the degree of coordination required and disruption tolerable during the upgrade process.
+
+### [Offline Migrations](/node/migrations/offline-migrations)
+
+Offline Migrations are quite simple and only requires coordination among the validators to take the network offline for a short period of time to upgrade the network. It uses network snapshots to bootstrap the new network with the same state as the existing network at the time of the snapshot.
+
+### [Zero Downtime Migrations](/node/migrations/zero-downtime-migrations)
+
+Zero-Downtime Migrations are a type of network migration that allows seamless upgrades to your Kwil network without causing any interruptions. With zero-downtime migrations, the network can be upgraded while the existing network continues to operate in parallel. During the migration process, the new network is set up and synchronized with the existing network, replicating any state changes that occur on the existing network. This ensures that the new network stays up to date with the latest changes while the migration is in progress. Once the migration is complete and the new network is fully operational, the old network can be decommissioned without any disruption to the network's users or services.
```

Comment:
I think it's a bit misleading to say "seamless" and "without causing any interruptions".  In this type of migration, we're starting a new chain.  This involves:

1. new chain
2. on-chain governance on old chain
3. new hosts, RPC endpoints, chain ID, etc

That's a lot of interruption and work.  This type of migration's strong suit is **zero-downtime** in the sense that the data will always be available as the old and new chains are running in parallel for a migration period.

Here's how I would rephrase this:

> In a zero-downtime migration, a new network is launched and run in parallel with the existing network for during a migration period.  This allows the data to be accessible through one network or the other throughout the process.
> 
> In this kind of migration, the net network is continually synchronizing with the original network, replicating any state changes that occur on the existing network.  During the migration period, users and applications should be updated to access data via the new network.  This ensures that the new network stays up to date with the latest changes while the migration is in progress. 
> 
> Once the ecosystem migration is complete and the migration period ends, the old network can be decommissioned without any disruption to the network's users or services.

Reply:
@brennanjl Regarding the above, one thing that I can't articulate clearly is what goes on with consumers during the migration period.  I alluded to an "ecosystem migration" in my rephrasing above, but I'm actually not certain what needs to be changed in apps or load balancers or client configs (e.g. RPC endpoints and chain ID settings etc).